### PR TITLE
feat(number-input): refactor  with testing

### LINKED ISSUE #8919: Refactor Number Input to use Sass modules
This issue is for updating the `Number Input` to Sass Modules in the `@carbon/styles` package. In this refactor, we will:

- [x] Copy over styles for for the component to `packages/styles/scss/components/number-input/_number-input.scss`
  - Any additional files will live in the folder `packages/styles/scss/components/number-input/`
- [x] Create an `_index.scss` at `packages/styles/scss/components/number-input/_index.scss`
- [x] Create a test file at `packages/styles/scss/components/__tests__/number-input-test.js`
- [x] add the storybook stories to @carbon/react

The `_index.scss` should:

- `@forward` the mixin and any variables from `_number-input.scss`
- Include the styles from `_number-input.scss` by calling the mixin

The `<component>-test.js` file should test the following:

- The Public API of the `number-input/_index.scss` file
- Any configuration options that are present for the file

The `_number-input.scss` will then need to be refactored to use the new import paths and conventions. You can find common recipes here: https://github.com/carbon-design-system/carbon/issues/8899#issuecomment-860887959

### PR DESCRIPTION
**Pull request by andreancardona**
Closes #8919 

- Refactors `NumberInput` to SASS modules 

### Testing: 

- [ ] run `yarn test --watch packages/styles/scss/components/__tests__/number-input-test.js`
- [ ] Make sure tests pass
- [ ] cd `packages/carbon-react` run `yarn storybook` to make sure none of the styles are broken

**REVIEWS** help me confirm the new variable name is in fact `$duration-fast-01` thank you :)

### CODE REVIEW

File: packages/styles/scss/components/number-input/_number-input.scss [hunk 1]
```diff
@@ -0,0 +1,479 @@
...
+//
+
+@use '../../colors' as *;
+@use '../../config' as *;
+@use '../../motion' as *;
+@use '../../spacing' as *;
+@use '../../theme' as *;
+@use '../../type' as *;
+@use '../../utilities/skeleton' as *;
+@use '../../utilities/button-reset';
```
Comment (sstrubberg) [inline packages/styles/scss/components/number-input/_number-input.scss:15]: Just confirming this doesn't need the `as *`;

Comment (andreancardona) [reply packages/styles/scss/components/number-input/_number-input.scss:15]: nope!

File: packages/styles/scss/components/number-input/_number-input.scss [hunk 2]
```diff
@@ -0,0 +1,479 @@
...
+  }
+
+  .#{$prefix}--number input[type='number'] {
+    @include type-style('body-short-01');
+    @include focus-outline('reset');
+
+    display: inline-flex;
+    width: 100%;
+    min-width: 9.375rem;
+    height: rem(40px);
```
Comment (joshblack) [inline packages/styles/scss/components/number-input/_number-input.scss:40]: I think you'll need to bring in the `convert` utilities package for this 👀

**Review by joshblack — CHANGES_REQUESTED** [review]
Just a small comment! 

Side-note, would it make sense to add the storybook stories to `@carbon/react` for this component to double-check that the styles look correct? 👀

**Comment by emyarod** [comment]
@joshblack should we be expecting storybook changes in other sass module reviews as well?

**Comment by joshblack** [comment]
@emyarod good question, might make sense to add to the checklist

File: packages/styles/scss/components/number-input/_number-input.scss (hunk 2 above)
Comment (andreancardona) [reply packages/styles/scss/components/number-input/_number-input.scss:40]: oh yes I'll update!

**Comment by andreancardona** [comment]
> @joshblack should we be expecting storybook changes in other sass module reviews as well?

Thank you for the feedback ya'll I'll make sure to add!

**Comment by andreancardona** [comment]
@joshblack @emyarod storybook was added for styles. Let me know what other feedback ya'll might have!

**Review by joshblack — APPROVED** [review]

**Review by sstrubberg — APPROVED** [review]
🔥

**Review by emyarod — APPROVED** [review]
looks good to me

**Pull request merged**